### PR TITLE
Properly handle string return from function

### PR DESCRIPTION
**Pull request by shukitchan**

### CODE REVIEW

**Comment by JosiahWI** [comment]
[approve ci rocky]

**Comment by JosiahWI** [comment]
[approve ci rocky]

**Review by bneradt — COMMENTED** [review]
Good fix.

**Comment by shukitchan** [comment]
[approve ci rocky]

**Comment by JosiahWI** [comment]
[approve ci rocky]

File: plugins/experimental/wasm/ats_context.h [hunk 1]
```diff
@@ -276,9 +276,8 @@ class Context : public ContextBase
   const char *
   getTransformResult(size_t *body_size)
   {
-    std::string s = transform_result_.get();
-    *body_size    = s.size();
-    return s.c_str();
+    *body_size = transform_result_.get().size();
```
Comment (cmcfarlen) [inline plugins/experimental/wasm/ats_context.h:279]: What if body_size is nullptr?

Comment (shukitchan) [reply plugins/experimental/wasm/ats_context.h]: good catch

File: plugins/experimental/wasm/ats_context.h [hunk 2]
```diff
@@ -276,9 +276,12 @@ class Context : public ContextBase
   const char *
   getTransformResult(size_t *body_size)
   {
-    std::string s = transform_result_.get();
-    *body_size    = s.size();
-    return s.c_str();
+    if (body_size) {
+      *body_size = transform_result_.get().size();
+      return transform_result_.get().c_str();
```
[the remark this answers is not in the document]
Comment (shukitchan) [reply plugins/experimental/wasm/ats_context.h:281]: Actually you are right. Let me redo this.

**Comment by shukitchan** [comment]
[approve ci rocky]

**Comment by JosiahWI** [comment]
[approve ci rocky]

**Comment by brbzull0** [comment]
Looks good to me. I'll let a non Yahoo to +1!
Thanks  @shukitchan

File: plugins/experimental/wasm/ats_context.h [hunk 3]
```diff
@@ -276,9 +276,12 @@ class Context : public ContextBase
   const char *
   getTransformResult(size_t *body_size)
   {
-    std::string s = transform_result_.get();
-    *body_size    = s.size();
-    return s.c_str();
+    if (body_size) {
```
Comment (traeak) [inline plugins/experimental/wasm/ats_context.h:279]: Check against nullptr please?

**Review by traeak — COMMENTED** [review]
Looks good: It looks like owned_data_str_ will make sure the result doesn't go out of scope.

**Review by traeak — APPROVED** [review]
looks good!

**Comment by cmcfarlen** [comment]
Cherry-picked to v10.0.x

**Pull request merged**